Update LDocs (addresses PR #9 and much more)

DecibillyJoel opened the pull request:
-Added player.stiffness, player.minimap, player.telestealth, player.keypressed, player.hat, player.camerax, player.cameray, recovery field for player.hurt and block.explode, player.autoup() and autodown() and etc, player.disableup() and etc, player.disablesuperjump() and enable, block.hurtplayer(), player.recoverytimer, player.hurttimer, player.frost, player.postGameEvent()
-Added usage to listeners
-Added read-only to all read-only fields
-Added laser and rocket launcher rotation accepted formats
-Added transferhit and transferfade fields to givelaser
-Replaced non-existent getPlayersCount() function with very existing playersCount field

DecibillyJoel committed:
Update LDocs

-Added player.stiffness, player.minimap, player.telestealth, player.keypressed, player.hat, player.camerax, player.cameray, recovery field for player.hurt and block.explode, player.autoup() and autodown() and etc, player.disableup() and etc, player.disablesuperjump() and enable, block.hurtplayer(), player.recoverytimer, player.hurttimer, player.frost, player.postGameEvent
-Added usage to listeners
-Added read-only to all read-only fields
-Added laser and rocket launcher rotation accepted formats
-Added transferhit and transferfade fields to givelaser
-Replaced non-existent getPlayersCount() function with very existing playersCount field

DecibillyJoel committed:
Fix minimap and teleStealth defaults, add tparam for player.hurt recovery

DecibillyJoel committed:
Removed reference to keycodes in keycode field

aromaa commented on player.lua:
```diff
@@ -211,3 +248,75 @@ end
+--- Is the specified key currently pressed?
+--- @tparam int keycode The specified key that is being checked.
+--- @return Returns true if pressed, otherwise false.
+--- @usage local Zpressed = tolua(player.keypressed(90))
```
```suggestion
--- @usage local Zpressed = tolua(player.keypressed(keys.Z))
```

aromaa commented on block.lua:
```diff
@@ -55,8 +55,18 @@ end
+--- @tparam number vel The strength of the impact that pushes the player. By default 0.3.
+--- @tparam number recovery The number of seconds the player takes to recover. By default 2.5.
+--- @tparam bool checkRecovery Prevents pushing the player if they have invulnerability received by recovering from damage. By default true.
+--- @usage block.hurt(3, 1.1)
```
```suggestion
--- @usage block.hurtplayer(3, 1.1)
```

aromaa commented on game.lua:
```diff
@@ -1,15 +1,20 @@
+playersCount = 1
+
 --- The event handler for game start.
+--- @usage game.start.addListener(function() player.removeitem() player.chat("bye bye halo :)") end)
```
These examples doesn't seem to provide much value.

```suggestion
--- @see eventlistener
```

DecibillyJoel committed:
Update player.lua to use keys.Z

I didnt even knows keys was a thing wow... well it doesn't appear in LDocs so I will try to make a commit that makes it appear

Co-authored-by: Joni Aromaa <me@joniaromaa.fi>

DecibillyJoel committed:
block.hurtplayer typo oops

Co-authored-by: Joni Aromaa <me@joniaromaa.fi>

DecibillyJoel committed:
@see instead of @usage

Didn't know @see was a thing

Co-authored-by: Joni Aromaa <me@joniaromaa.fi>

DecibillyJoel committed:
Got the rest of event listeners

aromaa reviewed (approved):
LGTM. Thanks.
